clients/src/main/java/org/apache/kafka/common/utils/Time.java

```diff
@@ -32,8 +33,29 @@
     public long nanoseconds();
 
     /**
+     * Returns time in milliseconds as does System.currentTimeMillis(),
+     * but uses elapsed time from an arbitrary epoch more like System.nanoTime().
+     * The difference is that if somebody changes the system clock,
+     * Time.currentElapsedTime will change but nanoTime won't.
+     * @return  The time in milliseconds from some arbitrary point in time.
+     */
+    public long currentElapsedTime();
```

Comment:
I filed KAFKA-2607 to modify the `Time` interface. In the meantime, maybe we can `nanoseconds` and `milliseconds` instead of introducing these methods? I can make the change if you agree.